polynote-kernel/src/main/scala/polynote/kernel/interpreter/scal/ScalaCompleter.scala

```diff
@@ -18,87 +21,46 @@ class ScalaCompleter[Compiler <: ScalaCompiler](
 
   import compiler.global._
 
-  def completions(cellCode: compiler.CellCode, pos: Int): UIO[List[Completion]] = {
+  def completions(cellCode: compiler.CellCode, pos: Int): URIO[Blocking, List[Completion]] = {
     val position = Position.offset(cellCode.sourceFile, pos)
+    //lazy val importInfos = (cellCode.wrappedImports ++ cellCode.compiledImports).map(new analyzer.ImportInfo(_, 1))
     def symToCompletion(sym: Symbol, inType: Type) = {
       val name = sym.name.decodedName.toString.trim // term names seem to get an extra space at the end?
-      val typ  = sym.typeSignatureIn(inType)
+      val typ  = sym.typeSignatureIn(inType).resultType
       val tParams = sym.typeParams.map(_.name.decodedName.toString)
       val vParams = TinyList(sym.paramss.map(_.map{p => (p.name.decodedName.toString: TinyString, compiler.unsafeFormatType(p.infoIn(typ)): ShortString)}: TinyList[(TinyString, ShortString)]))
       Completion(name, tParams, vParams, compiler.unsafeFormatType(typ), completionType(sym))
     }
 
-    def scopeToCompletions(scope: Scope, name: Name, inType: Type): List[Completion] = {
-      val isError = name.decoded == "<error>"
-      val matchingSymbols = scope.filter(isVisibleSymbol).collect {
-        case sym if isError || name.isEmpty || sym.name.startsWith(name) => sym.accessedOrSelf
-      }.toList
-
-      matchingSymbols.sortBy(_.name).map(symToCompletion(_, inType))
+    def memberToCompletion(result: Member) = {
+      symToCompletion(result.sym, result.tpe).copy(resultType = compiler.unsafeFormatType(result.tpe.resultType))
     }
 
     def completeSelect(tree: Select) = tree match {
-      case Select(qual, name) if qual.tpe != null =>
-        // start with completions from the qualifier's type
-        val scope = qual.tpe.members.cloneScope
-
-        // bring in completions from implicit enrichments
-        val context = locateContext(tree.pos).getOrElse(NoContext)
-        val ownerTpe = qual.tpe match {
-          case MethodType(List(), rtpe) => rtpe
-          case _ => qual.tpe
-        }
+      case sel@Select(qual, name) if qual.tpe != null =>
```

Comment:
So here, we can't use the presentation compiler's `completionsAt` to complete the `Select` tree. The reason is that the PC tries to get the currently selected `Name` part, but if that's empty (which is a common case, like when you type a dot after a term) then it causes an exception because of the way the PC tries to compute where the name part begins (it ends up being an invalid position). This is why we had to copy and paste a bunch of the implementation of this.